seqr/utils/elasticsearch/es_gene_agg_search.py

```diff
@@ -12,9 +14,11 @@ class EsGeneAggSearch(EsSearch):
     CACHED_COUNTS_KEY = None
 
     def aggregate_by_gene(self):
-        searches = {self._search}
+        searches = [self._search]
         for index_searches in self._index_searches.values():
-            searches.update(index_searches)
+            for index_search in index_searches:
+                if index_search not in searches:
+                    searches.append(index_search)
```

Comment:
Python 3 reports `TypeError: unhashable type: 'dict'` for `searches = {self._search}`. I changed to use a `list` rather than a `set`. The function should be equivalent. But I'm not sure whether it is a best way to do it.

Reply:
this was working in python 2 and as far as I know this should not have changed between python 2 and 3. Can you find out why it changed?

Reply:
In Python 2:
```
>>> from elasticsearch_dsl import Search
>>> a=Search()
>>> b={a}
>>> b
set([<elasticsearch_dsl.search.Search object at 0x1031bb990>])
```
However, in Python 3:
```
>>> from elasticsearch_dsl import Search
>>> a=Search()
>>> b={a}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'Search'
```
According to the [document](https://stackoverflow.com/questions/17493307/creating-set-of-objects-of-user-defined-class-in-python), the class must provide `__eq__()` and `__hash__()` methods to allow its objects to be set elements. `elasticsearch_dsl.Search` does has `__eq__()` but doesn't have `__hash_()`. I did an experiment by adding below to the Search class:
```
    def __hash__(self):
        return hash(str(self))
```
Then, `searches = {self._search}` can run without an error.
These seem about Python 3. Python 2 needs neither `__eq__()` nor `__hash__()` for the set.

Reply:
We don't like to change the elasticsearch library. So we have to avoid using a Search object as a set member.

Reply:
but if it is lacking an `__eq__` method, how do we know `index_search not in searches` will work at all?

Reply:
Hmm..., good question. But it has `__eq__` anyway.